File: crates/cxx-qt-lib/src/gui/qpixmap.cpp
// clang-format off
// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
// clang-format on
// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
//
// SPDX-License-Identifier: MIT OR Apache-2.0

#include "cxx-qt-lib/qpixmap.h"

namespace rust {
namespace cxxqtlib1 {

::std::unique_ptr<QPixmap>
qpixmapNew(int width, int height)
{
    return ::std::make_unique<QPixmap>(width, height);
}

} // namespace cxxqtlib1
} // namespace rust

File: crates/cxx-qt-lib/src/gui/qpixmap.rs
// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
//
// SPDX-License-Identifier: MIT OR Apache-2.0

#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        include!("cxx-qt-lib/qpixmap.h");
        type QPixmap;
        include!("cxx-qt-lib/qcolor.h");
        type QColor = crate::QColor;

        fn fill(self: Pin<&mut QPixmap>, color: &QColor);

        fn setDevicePixelRatio(self: Pin<&mut QPixmap>, ratio: f64);
    }

    #[namespace = "rust::cxxqtlib1"]
    unsafe extern "C++" {
        #[rust_name = "qpixmap_new"]
        fn qpixmapNew(width: i32, height: i32) -> UniquePtr<QPixmap>;
    }
}

pub use ffi::{QPixmap};

impl QPixmap {
    pub fn new(width: i32, height: i32) -> cxx::UniquePtr<Self> {
        ffi::qpixmap_new(width, height)
    }
}